### PR TITLE
Fix API docs

### PR DESCRIPTION
**Pull request by tayfun**
**Proposed changes**:

Fixes https://github.com/RasaHQ/rasa/issues/10148

- Updates `/model/train` endpoint in docs to only accept YAML input
- Updates `/model` endpoint to clarify model path

### CODE REVIEW

File: docs/static/spec/rasa.yml [hunk 1]
```diff
@@ -1052,7 +1053,7 @@ components:
         model_file:
           type: string
           description: Path to model file
-          example: "/models/20190512.tar.gz"
+          example: "/CHANGE-TO-DIR-ON-REMOTE-SERVER/models/20190512.tar.gz"
```
Comment (melindaloubser1) [inline docs/static/spec/rasa.yml:1056]:
```suggestion
          example: "/absolute-path-to-models-directory/models/20190512.tar.gz"
```

File: rasa/server.py [hunk 2]
```diff
@@ -1422,34 +1422,6 @@ def _test_data_file_from_payload(request: Request, temporary_directory: Path) ->
...
-        )
-
-    if "nlu" not in rjs and "stories" not in rjs:
-        raise ErrorResponse(
-            HTTPStatus.BAD_REQUEST,
-            "BadRequest",
-            "To train a Rasa model you need to specify at least one type of "
-            "training data. Add `nlu` and/or `stories` to the request.",
-            {"parameters": ["nlu", "stories"], "in": "body"},
-        )
```
Comment (melindaloubser1) [inline rasa/server.py:1441]: This is a functionality change, not just docs change - Can you tag an additional reviewer because I am not caught up on why this should or shouldn't happen

File: docs/static/spec/rasa.yml [hunk 3]
```diff
@@ -1407,19 +1408,40 @@ components:
...
 
-    JSONTrainingRequest:
+    YAMLTrainingRequest:
       type: object
       properties:
-        domain:
-          $ref: '#/components/schemas/DomainFile'
-        config:
-          $ref: '#/components/schemas/ConfigFile'
+        intents:
```
Comment (melindaloubser1) [inline docs/static/spec/rasa.yml:1414]: This section should include all properties (missing ones from the domain include actions, entities, e2e_actions, config) and be ordered (IMO) by config.yml, then domain.yml, then the data files

File: docs/static/spec/rasa.yml [hunk 4]
```diff
@@ -1860,3 +1881,50 @@ components:
...
           example: "Hello!"
+  examples:
+    YAMLTrainingRequest: |
+      intents:
+        - greet
+        - goodbye
+
+      responses:
+        utter_greet:
+        - text: "Hey! How are you?"
```
Comment (melindaloubser1) [inline docs/static/spec/rasa.yml:1892]: Same on this section: reorder by config.yml then domain.yml then data, and add all keys even if they're empty e.g. `actions: []`

**Review by melindaloubser1 — CHANGES_REQUESTED** [review]
Mostly concerned about the function removal - not sure that should be happening here?

File: rasa/server.py (hunk 2 above)
Comment (melindaloubser1) [reply rasa/server.py:1441]: Ope I see it auto tagged someone, so that's good

Comment (tayfun) [reply rasa/server.py:1441]: I'm removing code that's not used anymore. JSON was not supported, this function is not used currently. People have said docs had been stale for some time now.

File: docs/static/spec/rasa.yml (hunk 3 above)
Comment (tayfun) [reply docs/static/spec/rasa.yml:1457]: > This section should include all properties (missing ones from the domain include actions, entities, e2e_actions, config) and be ordered (IMO) by config.yml, then domain.yml, then the data files

@melindaloubser1 are you sure actions etc. would be handled here? I don't see it in stale docs either: https://rasa.com/docs/rasa/pages/http-api#operation/trainModel I've taken the example and schema from unit tests.

File: docs/static/spec/rasa.yml (hunk 4 above)
Comment (tayfun) [reply docs/static/spec/rasa.yml]: > Same on this section: reorder by config.yml then domain.yml then data, and add all keys even if they're empty e.g. `actions: []`

@melindaloubser1 I don't think actions are used here. I don't understand the ordering, why would it `session_config` come before others? Or do you mean `pipeline` comes first, then `policies`?

Comment (melindaloubser1) [reply docs/static/spec/rasa.yml]: For this particular example, there are no custom actions to pass, so you are correct. However, in a bot with custom actions, or slots, or entities, these sections would need to be included.

Re. the ordering: Sorry for the shorthand, I meant the contents of config.yml, so yes, pipeline + policies, then domain content (intents, entities, actions, e2e_actions, forms, slots, responses, config<- this is a key in the domain in 3.0, which is quite confusing) then content of nlu, rules and stories. It's not required to do it this way for the example to work, but will make it much easier for readers to understand the example

Comment (tayfun) [reply docs/static/spec/rasa.yml]: @melindaloubser1 Yeah I meant do you know custom actions are used if provided in request body like here? I don't see that in code, and I don't see them being tested in unit tests. I wouldn't be surprised if not all CLI options were supported here.

File: docs/static/spec/rasa.yml (hunk 3 above)
Comment (joejuzl) [reply docs/static/spec/rasa.yml:1457]: Isn't all of this covered by `#/components/schemas/DomainFile`?

File: docs/static/spec/rasa.yml [hunk 5]
```diff
@@ -1407,19 +1451,40 @@ components:
...
+            $ref: '#/components/schemas/TemplateDescription'
+        session_config:
+          description: Session configuration options
+          type: object
+          properties:
+            session_expiration_time:
+              type: integer
+            carry_over_slots_to_new_session:
+              type: boolean
+        pipeline:
```
Comment (joejuzl) [inline docs/static/spec/rasa.yml:1475]: Why the change from `#/components/schemas/ConfigFile`?

**Review by joejuzl — CHANGES_REQUESTED** [review]

File: docs/static/spec/rasa.yml (hunk 3 above)
Comment (tayfun) [reply docs/static/spec/rasa.yml:1457]: @joejuzl It's not under `domain` anymore, request body is flattened. This is why I've newly added this schema. Also, examples inside schemas do not work for any content type except for JSON in redocly, so `example` keys were removed and a separate example on the same level as schema was added. The other thing is I've simplified the example, previously there were 6 intents in the example request, I've added 2 to show how a list can be sent, I think people can extrapolate that and add more intents if needed.

File: docs/static/spec/rasa.yml (hunk 5 above)
Comment (tayfun) [reply docs/static/spec/rasa.yml:1475]: @joejuzl Similar to domain, request body is flattened, `config` key doesn't exist anymore.

Comment (joejuzl) [reply docs/static/spec/rasa.yml:1475]: is there no way to include the contents of `#/components/schemas/ConfigFile` at the top level, i.e. without it being under it's own key?

File: docs/static/spec/rasa.yml (hunk 4 above)
Comment (melindaloubser1) [reply docs/static/spec/rasa.yml]: They would have to be supported to train on stories which include custom actions I believe

File: docs/static/spec/rasa.yml (hunk 5 above)
Comment (tayfun) [reply docs/static/spec/rasa.yml:1475]: I've tried using `allOf` but couldn't get it to work. To test this, I've run `make livedocs` in top level and then added:
```
    TrainingSchema:
      allOf:
        - $ref: '#/components/schemas/DomainFile'
        - $ref: '#/components/schemas/ConfigFile'
```
to the bottom of the file and referenced it in requestBody as:
```
          application/yaml:
            schema:
              - $ref: '#/components/schemas/TrainingSchema'
```
I don't see schema and only see `any` in the docs output in that case. @joejuzl @melindaloubser1 do you know a better way?

Comment (tayfun) [reply docs/static/spec/rasa.yml:1475]: Well that second part's schema should not be a list:
```
         application/yaml:
            schema:
              $ref: '#/components/schemas/TrainingSchema'
```
Notice the dash is missing. This is still not showing the schema, but instead of `any`, I get `string (TrainingSchema)`

Comment (melindaloubser1) [reply docs/static/spec/rasa.yml:1475]: Unfortunately I haven't found another way either - Only experimented briefly but ran into the same issues you did @tayfun

File: rasa/server.py (hunk 2 above)
Comment (melindaloubser1) [reply rasa/server.py:1441]: Ah right, I see now there's no references left to this function anywhere 👍

**Review by joejuzl — APPROVED** [review]
Bit annoying we can't inline the other schemas... but at least it's correct now! 😁

**Review by melindaloubser1 — APPROVED** [review]
